ipapython/ipaldap.py

```diff
@@ -1749,3 +1752,239 @@ def get_ldap_uri(host='', port=389, cacert=None, ldapi=False, realm=None,
             return 'ldap://%s' % format_netloc(host, port)
         else:
             raise ValueError('Protocol %r not supported' % protocol)
+
+
+class CacheEntry:
+    def __init__(self, entry=None, attrs_list=None, exception=None,
+                 get_effective_rights=False, all=False):
+        self.entry = entry
+        self.attrs_list = attrs_list
+        self.exception = exception
+        self.all = all
+
+
+class LDAPCache(LDAPClient):
+    """A very basic LRU Cache using an OrderedDict"""
+
+    def __init__(self, ldap_uri, start_tls=False, force_schema_updates=False,
+                 no_schema=False, decode_attrs=True, cacert=None,
+                 sasl_nocanon=True, enable_cache=True, cache_size=100):
+
+        self.cache = OrderedDict()
+        self._cache_enabled = enable_cache
+        self._cache_size = cache_size
+        object.__setattr__(self, '_cache_misses', 0)
+        object.__setattr__(self, '_cache_hits', 0)
+        object.__setattr__(self, '_enable_cache', enable_cache)
+
+        object.__setattr__(self, '_cache_size', cache_size)
```

Comment:
Can only one of setting instance attributes be used here? I prefer `self.attr`, as I din't find `__setattr__` to be overriden.
Also, you are creating both `_cache_enabled` and `_enabled_cache`. Are both needed? I believe only one is required.

Reply:
I renamed the value at some point and didn't thoroughly move over to it, fixed.